src/deepsparse/transformers/utils/decoder_kv_cache.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict
+
+import numpy
+
+
+__all__ = ["DecoderKVCache"]
+
+
+# TODO: Dummy class just to enable testing, will be removed
+class KVCache:
+    def __init__(self):
+        pass
+
+
+class DecoderKVCache:
+    def __init__(self, use_deepsparse_cache: bool = False):
+        """
+        The goal this object is to handle the manipulation
+        of the key value cache.
+
+        :param use_deepsparse_cache: If set to True, the KVCache object
+            from the deepsparselib will be loaded as an attribute.
+            This object is used to handle the manipulation of the key
+            value buffers on the DeepSparse engine side.
+        """
+        # assuming that kv cache arrays are of shape
+        # [batch_size, num_heads, sequence_length, hidden_size]
+        self._sequence_axis = 2
```

Comment:
isn't this always the same for the engine? let's make this global to the file instead of a property